src/components/HeaderBar.js

```diff
@@ -46,39 +49,51 @@ class HeaderBar extends React.Component {
         window.removeEventListener( "resize", this.updateBarOpen );
         document.removeEventListener( 'click', this.handleClickOutside );
     }
+    
+
+    handleDarkModeChanged(){
```

Comment:
when dark mode changes, can just call the parent function to change the darkmode state in the App class, it will automaticaly re-render this HeaderBar.